src/widget-sources/widget-tileset-source-impl.ts

```diff
@@ -391,6 +393,78 @@ export class WidgetTilesetSourceImpl extends WidgetSource<WidgetTilesetSourcePro
     };
   }
 
+  async getAggregations({
+    aggregations,
+    filters,
+    filterOwner,
+    spatialFilter,
+  }: AggregationsRequestOptions): Promise<AggregationsResponse> {
+    const filteredFeatures = this._getFilteredFeatures(
+      spatialFilter,
+      filters,
+      filterOwner
+    );
+
+    if (!this._features.length) {
+      return {rows: []};
+    }
+
+    // Handle string-based aggregations
+    if (typeof aggregations === 'string') {
+      // For tilesets, string-based aggregations are not supported as they
+      // require SQL execution which is only available for remote sources
+      throw new Error('String-based aggregations not supported for tilesets');
+    }
+
+    // Handle array-based aggregations
+    const result: Record<string, number> = {};
+    const usedAliases = new Set<string>();
+
+    for (const {column, operation, alias} of aggregations) {
+      // Column is required except when operation is 'count'.
+      if ((column && column !== '*') || operation !== AggregationTypes.Count) {
+        assertColumn(this._features, column);
+      }
+
+      const aggregationKey = alias || `${operation}_${column}`;
+      const aliasKey = aggregationKey.toLowerCase();
+      if (usedAliases.has(aliasKey)) {
+        throw new Error(`Duplicate aggregation alias: ${aggregationKey}`);
+      }
+      usedAliases.add(aliasKey);
```

Comment:
If column is an empty string or '*' this might give unexpected results, I think. Do we know what alias the Model API chooses in those cases, and could we do the same locally?